integration-tests/hibernate-orm-panache/src/main/java/io/quarkus/it/panache/TestEndpoint.java

```diff
@@ -851,6 +859,47 @@ private void testPaging(PanacheQuery<Person> query) {
 
         Assertions.assertEquals(7, query.count());
         Assertions.assertEquals(3, query.pageCount());
+
+        // mix page with range
+        persons = query.page(0, 3).range(0, 1).list();
+        Assertions.assertEquals(2, persons.size());
+        Assertions.assertEquals("stef0", persons.get(0).name);
+        Assertions.assertEquals("stef1", persons.get(1).name);
+    }
+
+    private void testRange(PanacheQuery<Person> query) {
+        List<Person> persons = query.range(0, 2).list();
+        Assertions.assertEquals(3, persons.size());
+        Assertions.assertEquals("stef0", persons.get(0).name);
+        Assertions.assertEquals("stef1", persons.get(1).name);
+        Assertions.assertEquals("stef2", persons.get(2).name);
+
+        persons = query.range(3, 5).list();
+        Assertions.assertEquals(3, persons.size());
+        Assertions.assertEquals("stef3", persons.get(0).name);
+        Assertions.assertEquals("stef4", persons.get(1).name);
+        Assertions.assertEquals("stef5", persons.get(2).name);
+
+        persons = query.range(6, 8).list();
+        Assertions.assertEquals(1, persons.size());
+        Assertions.assertEquals("stef6", persons.get(0).name);
+
+        persons = query.range(8, 12).list();
+        Assertions.assertEquals(0, persons.size());
+
+        // mix range with page
+        try {
+            // invalid
+            query.range(0, 2).nextPage();
```

Comment:
Let's be thorough and test the rest of the methods that should throw.

Reply:
I added more test, and discover a missing check so worth it ;)